### PR TITLE
test: add Test Descriptions and Attributes for Improved Reporting and Analysis

### PR DESCRIPTION
**Pull request by adeldhis2**
This PR introduces a comprehensive update to our Cypress test suite, aiming to significantly improve test clarity and reporting capabilities within Report Portal. Prior to this update, understanding the nuances and purposes of our tests relied heavily on interpreting test names and code. Now, we've integrated detailed test attributes and descriptions, paving the way for more insightful and accessible test reporting and analysis.

### Key Enhancements:

**Standardized Test Attributes**: Implementation of mandatory 'Feature' and 'Action' attributes across all tests.
**Descriptive Test Details**: Each test now includes a clear, concise description, offering immediate context and intent without needing to delve into the code.
**Enhanced Reporting in Report Portal**: These additions significantly improve filtering options, reporting visibility, and analytical capabilities within Report Portal.

Included Screenshots from Report Portal:

_Results per DHIS2 Version:_

image

_Results per Features / DHIS2 Version: 40_

image

_Results per Action Attribute/ DHIS2 Version:40 / Feature: AlphanumericConditions_

image

_Test Descriptions / DHIS2 Version:40 / Feature: AlphanumericConditions / Action: VerifyOperators_

image

This update is a step towards more insightful and accessible testing practices, enabling us to better track, analyze, and understand our test suites within the DHIS2 ecosystem.

### CODE REVIEW

**Review by martinkrulltott — COMMENTED** [review]
Given that I'm missing the context for all of this, what the report portal is supposed to be used for, target audience, the ultimate goal/vision etc I fail to see how this change brings anything but redundancy to our code.

I'll use the following block as an example (the first line is the original code while the rest is added in this PR).

```
    it('legend is applied to negative values (per data item)', () => {
        cy.setTestDescription(
            'Validates the application of a legend to negative values in the visualization, per data item.'
        )
        cy.addTestAttributes([
            { key: 'feature', value: 'LegendSet' },
            { key: 'action', value: 'ApplyLegendToNegativeValues' },
        ])
```

`setTestDescription` seems to simply be a fleshed out version of the `it`, so we already have this information (albiet maybe not as easily readable, but readable enough imho). 

`key: 'feature', value: 'LegendSet'` is already provided by the name of the file (`legendSet.cy.js`). 

`{ key: 'action', value: 'ApplyLegendToNegativeValues' },` is already implied by the `it` and isn't human readable anyway, so I fail to see what value this really adds.

Basically, I don't mind the Report Portal collecting our test results, but having these 1300 lines added to our test files, which in turn has to be maintained, comes at a hefty price for no apparent use. Again, full disclosure, I'm very much missing the overall context here.

**Comment by adeldhis2** [comment]
Closing this PR as it was an exploratory effort to evaluate Report Portal features and their impact on our codebase. After discussion, we've decided not to proceed with these changes.
